OCCommunicationLib/OCCommunicationLib/OCOAuth2Authentication/OCCredentialsStorage.h

```diff
@@ -0,0 +1,38 @@
+//
+//  OCCredentialsStorage.h
+//  ownCloud iOS library
+//
+//  Created by Noelia Alvarez on 30/08/2017.
+//
+// Copyright (C) 2017, ownCloud GmbH.  ( http://www.owncloud.org/ )
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#import <Foundation/Foundation.h>
+#import "OCCommunication.h"
+
+@class OCCommunication;
+
+@protocol OCCredentialsStorageDelegate <NSObject>
+
+
+- (void)storeCurrentCredentialsOfSharedOCCommunication:(OCCommunication *)sharedOCCommunication;
```

Comment:
Would be easier if the method was something like ```(void)storeCredentials:(OCCredentialsDto *)credentials;``` . ```OCCredentialsStorage``` wouldn't depend on ```OCCommunication```, the lesser dependencies, the better.

Reply:
👍 You right, it is no longer needed the full communication object